mkdocs/docs/HPC/troubleshooting.md

```diff
@@ -1,6 +1,44 @@
+
+It could also be possible that more cores are simply not needed. You can test this by increasing the amount of cores and look at the speedup this increase gave you. For example, test with 2, 4, 16, 1/4 of all and all cores.
+
+Some **other reasons** why more cores could lead to no increase could be:
+
+- **Overhead:** When you use multithreading or multiprocessing, you can't expect that double the amount of cores will double the performance. This is due to the fact that it also needs time to create, manage and synchronize the threads/processes. When the overhead exceeds the speedup you receive by parallelization, it will not benefit the performance. For example, this can happen when you split you program in to many pieces to run in parallel. Creating a thread/process for this task will take longer than actually running the task itself.
+
+- **Amdahl's Law:** Amdahl's law is often used in parallel computing to predict the theoretical speedup when using multiple processors. The law states that "the overall performance improvement gained by optimizing a single part of a system is limited by the fraction of time that the improved part is actually used". For example, if a program needs 20 hours to complete using a single thread, but a one-hour portion of the program cannot be parallelized, therefore only the remaining 19 hours execution time can be parallelized. Regardless of how many threads are devoted to a parallelized execution of this program, the minimum execution time is always more than 1 hour. So when you reach this theoretical limit, more cores will not solve your problem.
+
+- **Resource contention:** When 2 or more threads/processes want to acces the same resources, they need to wait on each other. This is what we call resource contention. This results in that 1 thread/process will need to wait until the other one is is finished with this resource. When for example, each thread uses the same resource, it will definitely run slower than if it doesn't need to wait for the other threads to finish.
+
+- **Software Limitations:** It could also be that your software is just not really optimized for parallelization. A perfect example of a software that is not really optimized for multithreading is python. Although this has improved over the years. This is due to the fact that threads are implemented in a way that multiple threads can't run at the same time. This is due to the global interpreter lock (GIL). Instead of using multithreading in python to speedup a CPU bound program, you should use multiprocessing instead. This uses processes instead of actual threads. These can speedup you're CPU bound programs a lot more IN PYTHON than threads can do, although they are much les efficiënt to create. In other programming languages, you would probably wan't to use threads.
```

Comment:
```suggestion
- **Software Limitations:** It is possible that the software you are using is just not really optimized for parallelization. A perfect example of a software that is not really optimized for multi-threading is Python. (although this has improved over the years).
  This is due to the fact that in Python threads are implemented in a way that multiple threads can't run at the same time, due to the [global interpreter lock (GIL)](https://en.wikipedia.org/wiki/Global_interpreter_lock). 
  Instead of using multi-threading in Python to speedup a CPU bound program, you should use multi-processing instead, 
  which uses multiple processes (multiple instances of the same program) instead of multiple threads in a single program instance.
  Multiprocessing can speed up you're CPU bound programs a lot more *in Python* than threads can do,
  even though they are much less efficient to create.
  In other programming languages (which don't have a GIL), you would probably still want to use threads, however.